mvcc/kvstore.go

```diff
@@ -225,21 +227,18 @@ func (s *store) HashByRev(rev int64) (hash uint32, currentRev int64, compactRev
 	return hash, currentRev, compactRev, err
 }
 
-func (s *store) Compact(rev int64) (<-chan struct{}, error) {
-	s.mu.Lock()
+func (s *store) setCompactRev(rev int64) (time.Time, <-chan struct{}, error) {
```

Comment:
rename setCompactRev to updateCompactRev. it is not a simple set operation, since it checks the current rev and tries to do only update.